packages/react-components/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -236,7 +236,12 @@ const useMenuOpenState = (
   const { findFirstFocusable, findNextFocusable, findPrevFocusable } = useFocusFinders();
   const focusFirst = React.useCallback(() => {
     const firstFocusable = findFirstFocusable(state.menuPopoverRef.current);
-    firstFocusable?.focus();
+    if (firstFocusable) {
+      // For some reason, directly focusing the element here causes a nasty issue when opening the menu
+      // while inside an iframe with no scrollbars: the parent frame will be automatically scrolled to top
+      // focusing the menu item on the next repaint seems to fix the issue
+      requestAnimationFrame(() => firstFocusable.focus());
+    }
```

Comment:
I am not sure that `requestAnimationFrame()` is the best possible solution and can cause regressions in other places.

@ling1726 can you please check it?

Reply:
also please avoid using global `requestAnimationFrame` - always get one for the active `document` - not from global. @layershifter what is the correct way of getting it for v9?

Reply:
hello, do we have an update on this? Should we suggest a change?

Reply:
I'm still not sure what the root cause actually is, I tried to create a minimal repro and can't reproduce this issue. @tudorpopams  @radugroza  can you please check it out and see if it matches the scenario with the docsite ? 

https://codesandbox.io/s/jovial-dawn-cttm1t?file=/example.tsx

Reply:
Even more surprising the theme picker is actually the same menu component, but why is there no scroll when it is opened ?

![image](https://user-images.githubusercontent.com/20744592/184827204-ca4e99bd-5b19-4e59-bc7a-9c9fd74577d4.png)

Reply:
@ling1726 I forked the codesandbox sample and reproduced the issue here: https://codesandbox.io/s/cool-firefly-269epd?file=/example.tsx (the only difference is that the iframe loads the actual storybooks)

```html
<iframe src="https://react.fluentui.dev/iframe.html?viewMode=docs&id=components-button-menubutton--default" <!-- etc -->
```

Reply:
@radugroza  I'm not entirely we fully know what the root cause is, as I said earlier the theme picker is the same component, yet it doesn't trigger this scrolling. It seems that there is something storybook related that might be in play here

Reply:
@ling1726 actually the ThemeSwitcher does that too.

updated sample to reproduce it here: https://codesandbox.io/s/cool-firefly-269epd?file=/example.tsx
It reproduces if the top of the iframe is out of the viewport when opening the dropdown: 
<img width="952" alt="image" src="https://user-images.githubusercontent.com/659741/184897245-adaa8aa7-085c-48ac-ab6e-50043d065b80.png">

Reply:
That's interesting, I'll try to see if I can get a more isolated repro, I just want to rule out any storybook related weirdness with this

Reply:
Hi folks can we come to some resolution on this issue, please? Ling will be out of office for 2 weeks starting this Friday, so i am hoping to close this out soon 🙏

Reply:
So I created a developement build of the docsite with source maps https://v9storybook.z6.web.core.windows.net

I have noticed that when the popover is opened in an iframe, our [position fixed solution to avoid scroll jumps](https://github.com/microsoft/fluentui/blob/6853d071ca06579092a9cc3660fe9943479492d6/packages/react-components/react-positioning/src/usePositioning.ts#L267-L269) isn't working as expected.

When I open a menu without the iframe:

![image](https://user-images.githubusercontent.com/20744592/185646836-ec8a4934-5a9c-42a1-9b30-5a6947b6b4e5.png)


When I open a menu with the iframe:

![image](https://user-images.githubusercontent.com/20744592/185646561-9aefdefe-7d2a-4271-80e6-1250bfaf06f6.png)

Reply:
It seems that `position: fixed` inside the iframe seems to be relative to the iframe and not to the parent viewport. By setting `position:fixed` for the iframe it stops the scrolling but there are other issues... https://codepen.io/ling1726/pen/XWEONwB